hlsl-headers/include/vk/khr/cooperative_matrix.impl

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "common.hlsli"
+
+namespace vk {
+namespace khr {
+namespace internal {
+
+// These enums are expected to match the values in spirv.h
+// TODO: See if there is a way to use a spirv header to get these values
+// instead of redefining them.
+enum CooperativeMatrixOperand {
+  NoneKHR = 0x0,
+  MatrixASignedComponentsKHR = 0x1,
+  MatrixBSignedComponentsKHR = 0x2,
+  MatrixCSignedComponentsKHR = 0x4,
+  MatrixResultSignedComponentsKHR = 0x8,
+  MatrixAllSignedComponents = 0xF,
+  SaturatingAccumulationKHR = 0x10,
+  SaturatingAllSignedComponenets = 0x1F
+};
+
+enum class CooperativeMatrixLayout {
+  RowMajorKHR = 0x0,
+  ColumnMajorKHR = 0x1,
+};
+
+enum class CooperativeMatrixUse {
+  MatrixAKHR = 0,
+  MatrixBKHR = 1,
+  MatrixAccumulatorKHR = 2,
+};
+
+// Type-Declaration Instructions
+
+// TODO: make sure scope, rows, and cols can be specialization constants
+template <typename ComponentType, uint scope, uint rows, uint columns, uint use>
+using CooperativeMatrixKHR = vk::SpirvOpaqueType<
```

Comment:
Can we add `SPV_KHR_CooperativeMatrix` to a type? If not, we might need to add the ability to do that.

Reply:
Yes, seems like it would be the best option. The SPV_KHR_CooperativeMatrices can be added to the variable declarations, but it's not picked up by the compiler today.

Reply:
I added the ability to add the capability and extension to the member variable, and that is what I do now.